pkg/runtime/runtime.go

```diff
@@ -52,7 +55,35 @@ type kubernetes struct{}
 
 // NewRuntime - Initialize provider variable
 func NewRuntime(log *logging.Logger) {
-	Provider = &provider{log: log}
+	k8scli, err := clients.Kubernetes(log)
+	if err != nil {
+		log.Error(err.Error())
+		panic(err.Error())
+	}
+
+	// Identify which cluster we're using
+	restclient := k8scli.Client.CoreV1().RESTClient()
+	body, err := restclient.Get().AbsPath("/version/openshift").Do().Raw()
+
+	var cluster coe
+	switch {
+	case err == nil:
+		var kubeServerInfo kubeversiontypes.Info
+		err = json.Unmarshal(body, &kubeServerInfo)
+		if err != nil && len(body) > 0 {
+			log.Error(err.Error())
+			panic(err.Error())
+		}
+		log.Info("OpenShift version: %v", kubeServerInfo)
+		cluster = openshift{}
```

Comment:
Thinking about the future, can we make these a function that returns a `coe`?

Reply:
I was thinking that the coe interface would stay private so that only functions in the runtime pkg have access to openshift.go and kubernetes.go.  That way, the broker only asks for things it understands like "I want a new sandbox to securely create this apb" instead of "I want a few rbac rules, a new service account, and a new namespace".

What did you have in mind for the future with ```coe```?

Reply:
well I am not saying to make it public, I am talking about making an unexported helper method that encapsulates the retrieval of an actual `coe` structure.

This is for something like the network joining only for openshift and only when a certain plugin is installed. This means that we would want to do extra stuff while creating a `coe` and it was just a preemptive thing. Once we need it we can always add it as well.

Reply:
Ok I made an assumption of what you meant based on how I was using NewRuntime.  If I understand what you mean, you would like something similar to what we have with the kubernetes client? I'll post a revision with that assumption.